File: 모범 코드/8일차/8. 분해합.py
N = int(input())

flag = False
for i in range(N+1):
    j = i
    s = 0
    while j > 0:
        s += j%10
        j //= 10
    
    if s+i == N:
        print(i)
        flag = True
        break
        
if not flag: print(0)

File: 모범 코드/8일차/8. 분해합.java
import java.io.*;
import java.util.*;

public class main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());
        
        boolean flag=false;
        for(int i=1; i<=N; i++){
            int sum=0, j=i;
            while(j > 0){
                sum += j%10;
                j /= 10;
            }
            
            if(sum+i == N){
                System.out.print(i);
                flag = true;
                break;
            }
        }
        
        if(!flag) System.out.print(0);
    }
}